pkgs/by-name/po/poco/package.nix

```diff
@@ -54,20 +54,45 @@ stdenv.mkDerivation rec {
   MYSQL_DIR = libmysqlclient;
   MYSQL_INCLUDE_DIR = "${MYSQL_DIR}/include/mysql";
 
-  cmakeFlags = [
-    # use nix provided versions of sqlite, zlib, pcre, expat, ... instead of bundled versions
-    (lib.cmakeBool "POCO_UNBUNDLED" true)
-  ];
+  cmakeFlags =
+    let
+      excludeTestsRegex = lib.concatStringsSep "|" [
+        # These tests require running services, which the checkPhase is ill equipeed to provide
+        # TODO get them running in a nixosTest
+        "Redis"
+        "DataODBC"
+        "MongoDB"
+        "DataMySQL"
+        # network not accessible from nix sandbox
+        "NetSSL" # around 25 test failures
+        "Net" # could be made to work when public network access is patched out
+      ];
+    in
+    [
+      # use nix provided versions of sqlite, zlib, pcre, expat, ... instead of bundled versions
+      (lib.cmakeBool "POCO_UNBUNDLED" true)
+      (lib.cmakeBool "ENABLE_TESTS" true)
+      (lib.cmakeFeature "CMAKE_CTEST_ARGUMENTS" "--exclude-regex;'${excludeTestsRegex}'")
+    ];
 
-  patches = [
-    # Remove on next release
-    (fetchpatch {
-      name = "disable-included-pcre-if-pcre-is-linked-staticly";
-      # this happens when building pkgsStatic.poco
-      url = "https://patch-diff.githubusercontent.com/raw/pocoproject/poco/pull/4879.patch";
-      hash = "sha256-VFWuRuf0GPYFp43WKI8utl+agP+7a5biLg7m64EMnVo=";
-    })
-  ];
+  patches =
+    [
+      # Remove on next release
+      (fetchpatch {
+        name = "disable-included-pcre-if-pcre-is-linked-staticly";
+        # this happens when building pkgsStatic.poco
+        url = "https://patch-diff.githubusercontent.com/raw/pocoproject/poco/pull/4879.patch";
+        hash = "sha256-VFWuRuf0GPYFp43WKI8utl+agP+7a5biLg7m64EMnVo=";
+      })
+    ]
+    ++ lib.optional stdenv.hostPlatform.isDarwin ./disable-broken-tests-darwin.patch
+    ++ lib.optional stdenv.hostPlatform.isLinux ./disable-broken-tests-linux.patch;
```

Comment:
```suggestion
    ++ lib.optionals stdenv.hostPlatform.isDarwin [
    ./disable-broken-tests-darwin.patch
  ] ++ lib.optionals stdenv.hostPlatform.isLinux [
    ./disable-broken-tests-linux.patch;
  ];
```

Reply:
Sure fine, but could you perhaps give some rationale why this is better? Is this about the shorter line length? It seems that there is no real difference between `lib.optionals()` vs. `lib.optional()`?

Reply:
The brackets are explicit, it makes things easier to reason about, especially for newcomers in the nix language.